src/utils/validateInputWithRegex.ts

```diff
@@ -6,8 +6,8 @@ export const validateAndCleanRegexInput = (
   inputId: string,
   idRegex: string
 ): string => {
-  // set ID to all uppercase to remove case sensitivity
-  const id = inputId.toUpperCase();
+  // set ID to all uppercase to remove case sensitivity and removes trailing and leading white spaces
+  const id = inputId.trim().slice(0).toUpperCase();
```

Comment:
hmm whats the reason for including slice here? 
this might not be applicable, do take a look at the documentation: https://www.w3schools.com/jsref/jsref_slice_string.asp

Reply:
also! do add test cases for this cases, to ensure the behaviour is intended

Reply:
i added the slice since just to ensure we're returning a new copy of it, should I just remove it?

Reply:
added test cases!

Reply:
okay i think it's fine to include the slice. also do make sure to perform testing on the distributions where IDs are in regex form (i.e. FFTH, Haze) to ensure the behaviour is as intended!

Reply:
hello, I'd suggest to remove the slice, I came across this, was wondering why. Since `inputId` is a primitive string, don't need to worry about copying

![image](https://user-images.githubusercontent.com/577802/124877482-b3ba8f00-dffd-11eb-9b12-88924c665ac9.png)